app/src/main/java/org/listenbrainz/android/ui/navigation/SideNavigationBar.kt

```diff
@@ -0,0 +1,300 @@
+                                ?: item.iconUnselected),
+                            modifier = Modifier
+                                .size(24.dp)
+                                .padding(vertical = 4.dp),
+                            contentDescription = item.title,
+                            tint = MaterialTheme.colorScheme.onSurface
+                        )
+                        Text(
+                            text = item.title,
+                            color = MaterialTheme.colorScheme.onSurface,
+                            modifier = Modifier.padding(),
+                        )
+                    }
+
+                },
+                alwaysShowLabel = false,
+                selected = selected,
+                colors = NavigationRailItemDefaults.colors(
+                    indicatorColor = backgroundColor,
+                ),
+                onClick = {
+                    coroutineScope.launch {
+                        if (selected) {
+                            scrollToTop()
+                        }
+                        backdropScaffoldState.reveal()
+
+                        when (item.route) {
+                            AppNavigationItem.Profile.route -> {
+                                navController.navigate(AppNavigationItem.Profile.route + if (!username.isNullOrBlank()) "/${username}" else "") {
+                                    // Avoid building large backstack
+                                    popUpTo(navController.graph.findStartDestination().id) {
+                                        if (username.isNullOrBlank()) {
+                                            inclusive = true
+                                        }
+
+                                        saveState = true
+                                    }
+                                    // Avoid copies
+                                    launchSingleTop = true
+                                    // Restore previous state
+                                    restoreState = true
+                                }
+                            }
+
+                            else -> navController.navigate(item.route) {
+                                popUpTo(navController.graph.findStartDestination().id) {
+                                    saveState = true
+                                }
+                                // Avoid copies
+                                launchSingleTop = true
+                                // Restore previous state
+                                restoreState = true
+                            }
+                        }
+                    }
+                }
+            )
+        }
+        Spacer(modifier = Modifier.weight(1f))
+        BrainzNavigationMiniPlayer(
+            backdropScaffoldState = backdropScaffoldState,
+            currentlyPlayingSong = currentlyPlayingSong,
+        )
+    }
+}
+
+@Composable
+fun BrainzNavigationMiniPlayer(
```

Comment:
We need to re-use existing component somehow by making it flexible to resizing. Maintaining 2 components is a pain when it comes to updating something.